ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -225,7 +225,8 @@ async fn get_data_with_tasks(
         Ok(CacheCoverageStatus::CacheEvicted) => match transactions_count {
             None => MAX_FETCH_TASKS_PER_REQUEST,
             Some(transactions_count) => (transactions_count / TRANSACTIONS_PER_STORAGE_BLOCK)
-                .min(MAX_FETCH_TASKS_PER_REQUEST),
+                .min(MAX_FETCH_TASKS_PER_REQUEST)
+                .max(1),
```

Comment:
is it intended `min` > `max`?

Reply:
This is confusing and what lead to the bug in the first place lol. `min` takes the min 2 of numbers. `max` takes the max of 2 numbers. 
The `min` limits the max number of tasks to `MAX_FETCH_TASKS_PER_REQUEST` and the `max(1)` limits the minimum number of tasks to 1.

Reply:
Can we rewrite this in a clearer way? The new way is also pretty confusing w the .max and .min. I'd rather use if statement if it's already caused bugs.